integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/mutation/CarbonProjectForUpdateCommand.scala

```diff
@@ -59,6 +60,13 @@ private[sql] case class CarbonProjectForUpdateCommand(
       return Seq.empty
     }
     val carbonTable = CarbonEnv.getCarbonTable(databaseNameOp, tableName)(sparkSession)
+    columns.foreach { col =>
+      var dataType = carbonTable.getColumnByName(tableName, col).getColumnSchema.getDataType
+      if(dataType.getName.equals("STRUCT") || dataType.getName.equals("ARRAY")) {
```

Comment:
do not check for hard coded value, get the datatype and dicrectly check `isComplexType` and throw error